### ISSUE TITLE
Whatshap Haplotag error

### ISSUE THREAD

**Issue by Pharmacogenetecist** [whatshap/whatshap#347]
I am getting an error as follows:
"ERROR: whatshap error: invalid contig `chr1`"

Here is a copy of the first few lines of VCF file:
```
##FORMAT=<ID=PS,Number=1,Type=Integer,Description="Phase set identifier">
#CHROM	POS	ID	REF	ALT	QUAL	FILTER	INFO	FORMAT	BC66
chr22	42522074	.	C	A	975.8	PASS	BaseCalledReadsWithVariant=1566;BaseCalledFraction=0.84466;TotalReads=94;AlleleCount=2;SupportFraction=0.983309;SupportFractionByStrand=0.978078,0.990071;StrandSupport=1,0,52,41;StrandFisherTest=0;SOR=0.520961;RefContext=CTGGCCTATAG	GT	1/1
chr22	42522137	.	C	CTGT	9342.7	PASS	BaseCalledReadsWithVariant=1413;BaseCalledFraction=0.740178;TotalReads=658;AlleleCount=2;SupportFraction=0.948347;SupportFractionByStrand=0.968309,0.925523;StrandSupport=11,23,340,284;StrandFisherTest=17;SOR=0.523116;RefContext=GGTGACTGTGA	GT	1/1
```

**Comment by marcelm** [comment]
What’s the command you are trying to run?

**Comment by Pharmacogenetecist** [comment]
Haplotag
whatshap haplotag -o "x.haplotag.bam" --reference "hg19.fa" "66.phased.vcf.gz" "C66.renamed.bam" --output-haplotag-list HAPLOTAG

**Comment by Pharmacogenetecist** [comment]
> The VCF is missing parts of the header, which are probably relevant. Can you share the VCF and also the BAM files with us? (I may not be able to get back to you very soon, though.)

Yes Happy to, I will email you (marcel.martin@scilifelab.se). Don't worry, get back to me when possible, enjoy the festive season. Thanks

**Comment by marcelm** [comment]
Thanks for the data, that made it easy to find out what is going on. The problem is the same as the one reported in #337.

With the most recent WhatsHap version 1.2, the error message is different and more helpful:
```
Found 1 sample(s) in input VCF
Found 1 sample(s) in BAM file
ERROR: whatshap error: Input BAM/CRAM contains reads on contig 'chr1', but that contig does not exist in the VCF header. To bypass this check, use --skip-missing-contigs
```
Unfortunately, the option `--skip-missing-contigs`, which was added in WhatsHap 1.2, doesn’t work correctly, so for the moment, the workaround is to subset the BAM file, something like this:
```
samtools view -o BC66.renamed.onlychr22.bam BC66.renamed.bam chr22
samtools index BC66.renamed.bam
```
Please leave the issue open until I’ve fixed this.

And since you’re already the second person with this error message, perhaps I should reconsider making the `--skip-missing-contigs` behaviour the default.

**Comment by Pharmacogenetecist** [comment]
Thank you @marcelm that now works. Appreciate your help!

**Comment by marcelm** [comment]
Closing because `--skip-missing-contigs` was fixed in dfaac256ab754393dbf264144b45ff9cec80ba28.
